packages/svelte/src/compiler/migrate/index.js

```diff
@@ -34,6 +34,51 @@ class MigrationError extends Error {
 	}
 }
 
+/**
+ *
+ * @param {State} state
+ */
+function migrate_css(state) {
+	if (!state.analysis.css.ast?.start) return;
+	let code = state.str
+		.snip(state.analysis.css.ast.start, /** @type {number} */ (state.analysis.css.ast?.end))
+		.toString();
+	let starting = 0;
+
+	// since we already blank css we can't work directly on `state.str` so we will create a copy that we can update
+	const str = new MagicString(code);
+	while (!code.startsWith('</style>') && !!code.trim()) {
```

Comment:
Do we need this check? Just doing `while (code)` should be enough, because if there's whitespace at the end or if we're inside the `</style>` part we're not going to match anything of relevance anyway
```suggestion
	while (code) {
```

Reply:
Oh right i added this initially when it was running on the whole code.